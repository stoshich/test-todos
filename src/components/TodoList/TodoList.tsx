import React from 'react'
import { ITodo } from '../../types/data'
import TodoItem from '../TodoItem/TodoItem'

interface ITodoListProps {
    items: ITodo[];
    removeTodo: (id: number) => void;
    toggleTodo: (id: number) => void;
}

const TodoList: React.FC<ITodoListProps> = ({ items, removeTodo, toggleTodo }) => {
    return (
        <div className='todoList'>
            {items.map(item => (
                <TodoItem
                    key={item.id}
                    {...item}
                    removeTodo={removeTodo}
                    toggleTodo={toggleTodo}
                />))}
        </div>
    )
}

export default TodoList