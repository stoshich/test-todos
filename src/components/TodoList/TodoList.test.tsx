import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import TodoList from './TodoList';

const items = [
    { id: 1, title: 'todo 1', complete: false },
    { id: 2, title: 'todo 2', complete: true },
    { id: 3, title: 'todo 3', complete: false }
]
const removeTodo = jest.fn()
const toggleTodo = jest.fn()

describe('TodoList component', () => {
    it('Toggle todo works', () => {
        render(<TodoList
            items={items}
            removeTodo={removeTodo}
            toggleTodo={toggleTodo}
        />)
        userEvent.click(screen.getAllByRole('checkbox')[0])
        userEvent.click(screen.getAllByRole('checkbox')[1])
        expect(toggleTodo).toHaveBeenCalledTimes(2)
        expect(screen.getByText('todo 2')).toHaveClass('point__title completed')
    })
    it('Match snapshot', () => {
        render(<TodoList
            items={items}
            removeTodo={removeTodo}
            toggleTodo={toggleTodo}
        />)
        expect(screen.getAllByText(/todo/)).toMatchSnapshot()
    })
})