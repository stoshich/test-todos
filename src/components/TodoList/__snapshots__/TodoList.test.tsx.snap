// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList component Match snapshot 1`] = `
Array [
  <span
    class="point__title"
  >
    todo 1
  </span>,
  <span
    class="point__title completed"
  >
    todo 2
  </span>,
  <span
    class="point__title"
  >
    todo 3
  </span>,
]
`;
